src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -18,22 +17,18 @@ namespace NuGet.DependencyResolver
 {
     public static class ResolverUtility
     {
-        public static Task<GraphItem<RemoteResolveResult>> FindLibraryCachedAsync(
-            ConcurrentDictionary<LibraryRangeCacheKey, Task<GraphItem<RemoteResolveResult>>> cache,
+        public static async Task<GraphItem<RemoteResolveResult>> FindLibraryCachedAsync(
```

Comment:
I'd check all of these, and see where we can avoid adding the extra state machine.

Reply:
Done, I've updated all the places where it wasn't needed.  I wish we had an analyzer to tell us.